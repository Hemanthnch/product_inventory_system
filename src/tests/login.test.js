import React from 'react';
import { MemoryRouter as Router, withRouter } from 'react-router-dom'
import {mount} from 'enzyme';
import {shallow} from 'enzyme';
import Login from '../login'
import {configure} from 'enzyme'
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { render } from '@testing-library/react';
configure({adapter:new Adapter()});

it('Execute without fail',()=>{
  render(<div></div>)
})

it('Render',()=>{
  const wrapper=shallow(<Login />);
  const h=wrapper.find("h1");
  const result=h.text();
  expect(result).toBe("Login");
})

test('Login', () => {
    expect(true).toBeTruthy();
  });

it('Submit Function',()=>{
  const onSubmitFun=jest.fn();
  const wrapper=mount(<form onSubmit={onSubmitFun}></form>)
  const form=wrapper.find('form');
  form.simulate('submit');
  expect(onSubmitFun).toHaveBeenCalledTimes(1);
});
it('username ',()=>{
  const wrapper=mount(<input type="text" name="userName"/>);
  const input=wrapper.find('input');
  expect(input).toHaveLength(1);
  expect(input.prop('type')).toEqual('text');
  expect(input.prop('name')).toEqual('userName');
})
it('Password ',()=>{
  const wrapper=mount(<input type="password" name="password"/>);
  const input=wrapper.find('input');
  expect(input).toHaveLength(1);
  expect(input.prop('type')).toEqual('password');
  expect(input.prop('name')).toEqual('password');
})
it('Button ',()=>{
  const wrapper=mount(<button type="submit">Login</button>);
  const input=wrapper.find('button');
  const result=input.text();
  expect(result).toBe('Login')
  expect(input.prop('type')).toEqual('submit');
})
test('matches Snapshot',()=>{
  const tree=renderer.create(<Router><Login /></Router>).toJSON()
  expect(tree).toMatchSnapshot();
})