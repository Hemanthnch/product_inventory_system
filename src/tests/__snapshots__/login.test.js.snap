// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "backgroundColor": "blue",
        "color": "#f2f2f2",
        "font": "17px",
        "overflow": "hidden",
      }
    }
  >
    <h1
      style={
        Object {
          "color": "black",
        }
      }
    >
      Product Inventory System
    </h1>
    <ul
      style={
        Object {
          "listStyleType": "none",
        }
      }
    >
      <li
        style={
          Object {
            "backgroundColor": "cyan",
            "color": "black",
            "display": "inline",
            "float": "right",
            "margin": "5px",
            "padding": "5px",
          }
        }
      >
        <a
          href="/signup"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          Sign Up
        </a>
      </li>
      <li
        style={
          Object {
            "backgroundColor": "cyan",
            "color": "black",
            "display": "inline",
            "float": "right",
            "margin": "5px",
            "padding": "5px",
          }
        }
      >
        <a
          href="/login"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          Login
        </a>
      </li>
    </ul>
  </div>
  <form
    style={
      Object {
        "backgroundColor": "white",
        "border": "15px solid blueviolet",
        "color": "Black",
        "display": "inline-block",
        "margin": "20px",
        "padding": "10px",
        "width": "400px",
      }
    }
  >
    <h1>
      Login
    </h1>
    <h3>
      Enter the User Name 
    </h3>
    <input
      id="userName"
      onBlur={[Function]}
      onChange={[Function]}
      required={true}
      type="text"
    />
    <h3>
      Enter the Password 
    </h3>
    <input
      id="password"
      onBlur={[Function]}
      onChange={[Function]}
      required={true}
      type="password"
    />
    
    <br />
    <br />
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "cyan",
        }
      }
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;

exports[`mathes Snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "backgroundColor": "blue",
        "color": "#f2f2f2",
        "font": "17px",
        "overflow": "hidden",
      }
    }
  >
    <h1
      style={
        Object {
          "color": "black",
        }
      }
    >
      Product Inventory System
    </h1>
    <ul
      style={
        Object {
          "listStyleType": "none",
        }
      }
    >
      <li
        style={
          Object {
            "backgroundColor": "cyan",
            "color": "black",
            "display": "inline",
            "float": "right",
            "margin": "5px",
            "padding": "5px",
          }
        }
      >
        <a
          href="/signup"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          Sign Up
        </a>
      </li>
      <li
        style={
          Object {
            "backgroundColor": "cyan",
            "color": "black",
            "display": "inline",
            "float": "right",
            "margin": "5px",
            "padding": "5px",
          }
        }
      >
        <a
          href="/login"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          Login
        </a>
      </li>
    </ul>
  </div>
  <form
    style={
      Object {
        "backgroundColor": "white",
        "border": "15px solid blueviolet",
        "color": "Black",
        "display": "inline-block",
        "margin": "20px",
        "padding": "10px",
        "width": "400px",
      }
    }
  >
    <h1>
      Login
    </h1>
    <h3>
      Enter the User Name 
    </h3>
    <input
      name="username"
      onBlur={[Function]}
      onChange={[Function]}
      required={true}
      type="text"
    />
    <h3>
      Enter the Password 
    </h3>
    <input
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      required={true}
      type="password"
    />
    
    <br />
    <br />
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "cyan",
        }
      }
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
